### DISCUSSION TITLE
Sub directory for queries folder client

### DISCUSSION THREAD

**Discussion by robp94** [ExpediaGroup/graphql-kotlin#1806] [Q&A]
Is it possible to have subdirectories in the queries folder for the client generator?

Currently, we use this as config `<queryFileDirectory>${project.basedir}/src/main/resources/clients/graphql/queries</queryFileDirectory>`

If the queries are in exactly this folder it is working, if we add a new folder inside the queries folder they are ignored.

**Comment by dariuszkuc** [comment]
Hello 👋 
Logic to locate the query files is currently non-recursive. While not a perfect solution, instead of a directory your can also explicitly specify a list of files (which may be troublesome if you have tons of files).

Using Gradle you could easily script it out but for Maven explicit files are the only option.

You could open a feature request for this but it is unlikely it will be picked up anytime soon. We are always open to contributions though (it does seem like it would be a rather simple change in [Maven](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/plugins/graphql-kotlin-maven-plugin/src/main/kotlin/com/expediagroup/graphql/plugin/maven/GenerateClientAbstractMojo.kt#L152) mojo and [Gradle](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/plugins/graphql-kotlin-gradle-plugin/src/main/kotlin/com/expediagroup/graphql/plugin/gradle/tasks/AbstractGenerateClientTask.kt#L151) task).

Thanks,
Derek

**Reply by robp94** [reply]
Thanks for the quick response, I will try to make a pr in the near future.
